src/main/java/org/example/controllers/AuthController.java

```diff
@@ -0,0 +1,38 @@
+package org.example.controllers;
+
+import io.swagger.annotations.Api;
+import lombok.AllArgsConstructor;
+import org.example.exception.AuthDaoException;
+import org.example.exception.InvalidCredentialsException;
+import org.example.models.LoginRequest;
+import org.example.services.AuthService;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Api("Kainos Auth API")
+@Path("/api/auth") @AllArgsConstructor
+public class AuthController {
+    AuthService authService;
+
+    @POST
+    @Path("/login")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response login(final LoginRequest loginRequest) {
+        try {
+            return Response
+                    .ok()
+                    .entity(authService.login(loginRequest))
+                    .build();
+        } catch (AuthDaoException e) {
+            return Response.serverError().entity(e.getMessage()).build();
+        } catch (InvalidCredentialsException e) {
+            return Response
+                    .status(Response.Status.BAD_REQUEST)
```

Comment:
Should this be a bad request or should it be something else? From my understanding we return bad request when the user gives us something not in the right format (characters when something should be numbers, etc.). Is this checking that or is this checking if the username and password exist in the db

Reply:
That's a great point. I think it should a 401 unauthorized status code. [HTTP Status Codes](https://auth0.com/blog/forbidden-unauthorized-http-status-codes/). Making the changes now!